diff --git a/lib/src/thread.c b/lib/src/thread.c
index 0d6adf0..eb98ea1 100644
--- a/lib/src/thread.c
+++ b/lib/src/thread.c
@@ -13,7 +13,7 @@
 #include <switch.h>
 #endif
 
-#if defined(__ANDROID__)
+#if defined(__ANDROID__) || defined(__EMSCRIPTEN__)
 struct chiaki_timedjoin_ctx
 {
 	pthread_t target;
@@ -309,7 +309,7 @@ CHIAKI_EXPORT ChiakiErrorCode chiaki_thread_timedjoin(ChiakiThread *thread, void
 		return CHIAKI_ERR_THREAD;
 	if(retval)
 		*retval = thread->ret;
-#elif defined(__ANDROID__)
+#elif defined(__ANDROID__) || defined(__EMSCRIPTEN__)
 	// Android Bionic lacks pthread_clockjoin_np/pthread_timedjoin_np.
 	// Use a helper thread + condvar to implement timed join portably.
 	struct chiaki_timedjoin_ctx *ctx = calloc(1, sizeof(*ctx));
