diff --git a/third-party/gf-complete/src/gf_cpu.c b/third-party/gf-complete/src/gf_cpu.c
index f65131f..3200355 100644
--- a/third-party/gf-complete/src/gf_cpu.c
+++ b/third-party/gf-complete/src/gf_cpu.c
@@ -171,10 +171,9 @@ void gf_cpu_identify(void)

 #else // defined(__arm__) || defined(__aarch64__)

-int gf_cpu_identify(void)
+void gf_cpu_identify(void)
 {
     gf_cpu_identified = 1;
-    return 0;
 }

 #endif
